stm/stm.ml

```diff
@@ -114,10 +113,12 @@ let vernac_interp ?proof id ?route { verbose; loc; expr } =
   end
 
 (* Wrapper for Vernac.parse_sentence to set the feedback id *)
-let vernac_parse ?newtip ?route eid s =
-  let feedback_id = 
-    if Option.is_empty newtip then Edit eid
-    else State (Option.get newtip) in
+
+(* We report feedback for parsing errors with the stateId the parsing
+   is being attempted on. This is more convenient for IDEs, as the can
+   manage the state in an easier way only caring about stateIds.
+*)
+let vernac_parse ?route feedback_id eid s =
```

Comment:
In fact we could remove the feedback use here altogether, what useful feedback can a non-parseable statement produce?